### PR TITLE
[AV-76499] System events and activity logs

### PR DESCRIPTION
**Pull request by aniket-Kumar-c**
## Jira

* [AV-76499](https://couchbasecloud.atlassian.net/browse/AV-76499)

## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

### How was this change tested and do you have evidence? _**(REQUIRED: Select at least 1)**_

- [x] Manually tested
- [ ] Unit tested
- [ ] Acceptance tested
- [ ] Unable to test / will not test (Please provide comments in section below)

### Testing

<details open>
  <summary>Testing</summary>
  
</details>

## Required Checklist:

- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if required)
- [x] I have run make fmt and formatted my code

## Further comments

### CODE REVIEW

**Comment by aniket-Kumar-c** [comment]
Validation are added in respective readme file

**Review by PaulomeeCb — COMMENTED** [review]
Changes lgtm! just a question, do we need to update the README of the getting started folder? Or will it remain separately as we cant have the events along with the other resources (i.e. like how we update the getting_started` variables.tf` and `terraform.template.tfvars` for resources together)

**Review by PaulomeeCb — APPROVED** [review]
+1 , just remove the lint errors. Thanks for change!

File: internal/datasources/project_events.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		to             string
+		page           int
+		perPage        int
+		sortBy         string
+		sortDir        string
+		err            error
+	)
+	queryParam := make(map[string][]string)
+
+	if !state.ClusterIds.IsNull() && !state.ClusterIds.IsUnknown() {
```
Comment (PaulomeeCb) [inline internal/datasources/project_events.go:70]: Nit: Can we have a separate function call like `Validate` for these null and unknown checks?

File: internal/datasources/events.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+		to             string
+		page           int
+		perPage        int
+		sortBy         string
+		sortDir        string
+		err            error
+	)
+	queryParam := make(map[string][]string)
+
+	if !state.ProjectIds.IsNull() && !state.ProjectIds.IsUnknown() {
```
Comment (PaulomeeCb) [inline internal/datasources/events.go:70]: Nit: maybe we can have a separate function like `Validate` for these checks?

**Review by PaulomeeCb — APPROVED** [review]
+1 LGTM

**Review by SaicharanCB — APPROVED** [review]

**Pull request merged**